
// Protobufs used in tests.

syntax = "proto2";

import "grr/proto/export.proto";
import "grr/proto/jobs.proto";
import "grr/proto/semantic.proto";


message PriorityFlowArgs {
  optional string msg = 1;
}

message ClientActionRunnerArgs {
  optional string action = 1 [(sem_type) = {
      description: "Action to run.",
    }];
}

message BadArgsFlow1Args {
  optional PathSpec arg1 = 1;
}

message SendingFlowArgs {
  optional uint64 message_count = 1;
}

message DummyCronHuntOutputPluginArgs {
  optional string output_path = 1;
  optional string collection_name = 2;
}

message RecursiveTestFlowArgs {
  optional uint64 depth = 1;
}

message MultiGetFileTestFlowArgs {
  optional uint64 file_limit = 1 [(sem_type) = {
      description: "The number of files to retrieve.",
    }, default=3];
}

// Used for DataAgnosticConverter tests.
message DataAgnosticConverterTestValue {
  optional string string_value = 1;
  optional uint64 int_value = 2;
  optional bool bool_value = 3;

  repeated string repeated_string_value = 4;
  optional ExportedMetadata message_value = 5;

  enum EnumOption {
    OPTION_1 = 0;
    OPTION_2 = 1;
  }
  optional EnumOption enum_value = 6 [default = OPTION_1];
  optional EnumOption another_enum_value = 7 [default = OPTION_2];

  optional string urn_value = 8 [(sem_type) = {
      type: "RDFURN",
    }];
  optional uint64 datetime_value = 9 [(sem_type) = {
      type: "RDFDatetime",
    }];
}

// Used for DataAgnosticConverter tests.
message DataAgnosticConverterTestValueWithMetadata {
  optional uint64 metadata = 1;
  optional string value = 2;
}

// For forms_test.py to test that default protobuf values are rendered
// correctly.
message DefaultArgsTestFlowArgs {
  enum EnumOption {
    OPTION_1 = 0;
    OPTION_2 = 1;
  }
  optional string string_value = 1;
  optional uint64 int_value = 2;
  optional bool bool_value = 3;
  optional EnumOption enum_value = 4;

  optional string string_value_with_default = 5 [default = "default string"];
  optional uint64 int_value_with_default = 6 [default = 42];
  optional bool bool_value_with_default = 7 [default = true];
  optional EnumOption enum_value_with_default = 8 [default = OPTION_2];
}